Lib/subprocess.py

```diff
@@ -53,6 +53,14 @@
 import contextlib
 from time import monotonic as _time
 
+try:
+    import pwd
+except ImportError:
+    pwd = None
+try:
+    import grp
+except ImportError:
+    grp = None
```

Comment:
You allow `pwd` and `grp` to be missing, but use them unconditionally below. The same problem exists in tests as well.

Reply:
OK, I will make it raise ValueError out if `pwd` or `grp` is missing and there is an attempt to use a named user or group. I will allow integers only, since that can still work.